.github/scripts/filter_changed_files_go_test.sh

```diff
@@ -1,5 +1,7 @@
 #!/bin/bash
 
+set -euo pipefail
+
 # Get the list of changed files
 files_to_check=$(git diff --name-only origin/$GITHUB_BASE_REF)
```

Comment:
Oh, I'm betting `$GITHUB_BASE_REF` isn't set on merges to main. In that case, we should just diff HEAD vs HEAD^ I guess?